src/java.base/share/classes/jdk/internal/constant/ConstantUtils.java

```diff
@@ -47,6 +49,59 @@ public final class ConstantUtils {
     /** No instantiation */
     private ConstantUtils() {}
 
+    // Note:
+    // Non-JDK users should create their own utilities that wrap
+    // {@code .describeConstable().orElseThrow()} calls;
+    // these xxDesc methods has undefined and unsafe exceptional
+    // behavior, so they are not suitable as public APIs.
+
+    /**
+     * Creates a ClassDesc from a Class object, requires that this class
+     * can always be described nominally, i.e. this class is not a
+     * hidden class or interface or an array with a hidden component
+     * type.
+     */
+    public static ClassDesc classDesc(Class<?> type) {
+        if (type.isPrimitive()) {
+            return Wrapper.forPrimitiveType(type).classDescriptor();
+        }
+        return ReferenceClassDescImpl.ofValidated(type.descriptorString());
+    }
+
+    /**
+     * Creates a MethodTypeDesc from a MethodType object, requires that
+     * the type can be described nominally, i.e. all of its return
+     * type and parameter types can be described nominally.
+     */
+    public static MethodTypeDesc methodDesc(MethodType type) {
```

Comment:
Please name these methods `methodTypeDesc`, since we also have the `Method` type.

Reply:
Thanks for the suggestion. Done.